frontend/packages/vkt/src/redux/sagas/clerkExamEventOverview.ts

```diff
@@ -58,6 +64,51 @@ function* updateClerkExamEventDetailsSaga(
   }
 }
 
+function* updateClerkEnrollmentStatusSaga(
+  action: PayloadAction<{
+    statusUpdate: ClerkEnrollmentStatusUpdate;
+    examEvent: ClerkExamEvent;
+  }>
+) {
+  const { statusUpdate, examEvent } = action.payload;
+  try {
+    const apiResponse: AxiosResponse<ClerkEnrollmentResponse> = yield call(
+      axiosInstance.put,
+      `${APIEndpoints.ClerkEnrollment}/status`,
+      statusUpdate
+    );
+    const enrollment = apiResponse.data;
+
+    const updatedEnrollment = SerializationUtils.deserializeClerkEnrollment(
+      apiResponse.data
+    );
+
+    const updatedEnrollments = [...examEvent.enrollments];
+    const idx = updatedEnrollments.findIndex((e) => e.id === enrollment.id);
+    updatedEnrollments[idx] = updatedEnrollment;
+
+    const updatedParticipants = updatedEnrollments.filter((enrollment) =>
+      [EnrollmentStatus.PAID, EnrollmentStatus.EXPECTING_PAYMENT].includes(
+        enrollment.status
+      )
+    ).length;
```

Comment:
Ei taida olla tarpeellinen koodiblokki kun participants ei ole osana `ClerkExamEvent` modelia. Vastaavaan liittyen voisi `DraftClerkExamEvent`:stä poistaa Omit-osiosta tuon `participants` avaimen.

Reply:
Tosin sellainen toinen havainto, että `upsertExamEvents` alla tehty oletus, että osallistujien määrä koostuu vain sellaisista ilmoittautumisista joiden tyyppi on PAID. Siellä pitäisi ottaa huomioon tosiaan myös nämä EXPECTING_PAYMENT alla olevat.

Reply:
Joo, tuo participants logiikka oli tosiaan tuolla reducerin puolella. Poistin sagasta ja upsertExamEvents myös puhdistettu ylimääräinen data (version, enrollments) niin ei kontaminoidu noissa päivityksissä. Eihän tuo vaikuta sovelluksen toimintaan mutta jälkikäteen järkevämmän näköinen katella kun data pysyy samanmuotoisena.

Hävitin myös `participants`:in draftista